rules/utils/get-number-literal-type.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+/**
+ * Return the type of the number in the node
+ * @param {Node} node - The node to check. Its type must be "Literal", and its value must be a number.
+ * @returns {undefined|"bigint"|"hexadecimal"|"binary"|"octal"|"number"}
+ */
+function getNumberLiteralType(node) {
+	if (!node || node.type !== 'Literal' || (typeof node.value !== 'number' && !node.bigint)) {
+		return;
+	}
+
+	if (node.raw.startsWith('0x')) {
```

Comment:
`0XdEED_BEEF`

Reply:
Oh, I did not know that prefixes could be uppercase too... That raises another issue: if someone uses uppercase, then the autofix will fix to lowercase... Is this an issue or is it ok?

Reply:
I prefer handle uppercase in this rule, people may disable `number-literal-case`, especially with `eslint-config-prettier` it's disabled by default https://github.com/prettier/eslint-config-prettier/blob/5d479e943313d078129ad3308bf4d16f69eeda53/unicorn.js#L6